_posts/2018-11-16-ruby-chef-hashicorp-vault.md

```diff
@@ -0,0 +1,178 @@
+---
+layout:     post
+title:      Ruby vs chef and the siege of Hashicorp Vault
+date:       2018-11-16
+summary:    How I discovered the dark secret behind chef and used Ruby to harness the power of Hashicorp Vault
+category:   Devops
+tags:       Hashicorp,chef,Ruby
+author:     andrei_sandu
+---
+## Hashicorp Vault - the stronghold
+
+It is a company focus to ensure our secrets are secure, and the solution adopted sits with [Hashicorp Vault](https://www.vaultproject.io/).
+Great work has been put by the Delivery Engineering team within Sky Betting & Gaming to formalise the process of storing and retrieving
+secrets in a seamless way, while maintaining security and ensuring compatibility with Chef and the use cases and business needs of the
+company. When dealing with Chef runs, which happen all the time on all of our systems, the main concern is the secrets becoming exposed
+by virtue of being loaded on disk during the compile phase. The solution we have for this restricts loading secrets to runtime only.
+The main side effect of this is that it has limited the extent to which one can perform complex logic with Hashicorp Vault secrets. The first
+part of this blog post focuses on the different options available to the DevOps engineers when it comes to Hashicorp Vault, the advantages and
+disadvantages of the different ways one can interact with Vault, and a personal note on responsibility and freedom when using secure data.
+
+## Restricting access for Chef and the problems it creates
+
+There are many different ways of interacting with secrets in Hashicorp Vault. When it comes to Chef cookbooks, the Sky Betting and Gaming way of doing it is as follows:
+ 1. _include_recipe 'sbg_vault::chef_auth'_ - make sure your recipe includes the wrapper cookbook created to enable our chef runs to access Hashicorp Vault
+ 2. use the _sbg_vault_secret_ resource - this resource was designed to be easy to use for the basic needs of a cookbook and at the same time ensure nobody
```

Comment:
Quite a few uses of underscores for inline code, which should be styled using backticks instead, e.g. `sbg_vault_secret`

Reply:
I see. will ammend